docs/spec_draft.md

```diff
@@ -2279,6 +2280,85 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.triangular_solve
+
+### Semantics
+
+Solves systems of linear equations of a batch of matrices with lower or upper
+triangular coefficients by forward- or back-substitution.
+
+More formally, given `a` and `b`, `result[i0, ..., iR-3, :, :]` is the solution
+to `op(a[i0, ..., iR-3, :, :]) * x = b[i0, ..., iR-3, :, :]` when `left_side` is
+`true` or `x * op(a[i0, ..., iR-3, :, :]) = b[i0, ..., iR-3, :, :]` when
+`left_side` is `false`, solving for the variable `x` where `op(a)` is either
+`op(a) = a`, or `op(a) = Transpose(a)`, or `op(a) = Conj(Transpose(a))`.
+
+Input data is read only from the lower triangle of `a`, if `lower` is `true` or
+upper triangle of `a`, otherwise. Output data is returned in the same triangle;
+the values in the other triangle are implementation-defined.
+
+If `unit_diagonal` is true, the diagonal elements of `a` are assumed to be 1 and
+not accessed.
+
+`transpose_a` can be one of the following:
+  * `NO_TRANSPOSE`: Perform operation using `a` as-is.
+  * `TRANSPOSE`: Perform operation on transpose of `a`.
+  * `ADJOINT`: Perform operation on conjugate transpose of `a`.
+
+### Inputs
+
+| Name            | Type                                               |
+|-----------------|----------------------------------------------------|
+| `a`             | tensor of floating-point or complex type           |
+| `b`             | tensor of floating-point or complex type           |
+| `left_side`     | constant of type `i1`                              |
+| `lower`         | constant of type `i1`                              |
+| `unit_diagonal` | constant of type `i1`                              |
+| `transpose_a`   | enum of `NO_TRANSPOSE`, `TRANSPOSE`, and `ADJOINT` |
+
+### Outputs
+
+| Name     | Type                                     |
+|----------|------------------------------------------|
+| `result` | tensor of floating-point or complex type |
+
+### Constraints
+
+  * (C1) `a` and `b` have the same element type and rank $\ge$ 2.
+  * (C2) dim(`a`, -2) = dim(`a`, -1).
+  * (C3) dim(`a`, `i`) $=$ dim(`b`, `i`) for all `i` $\in$ [0, R-3]. If
+    `left_size = true`, dim(`b`, -2) $=$ dim(`a`, -1) and dim(`b`, -1) $=$
+    dim(`a`, -1) otherwise.
+  * (C4) `b` and `result` have the same type.
+
+### Examples
+
+```mlir
+// %a = [
+//       [1, 0, 0],
+//       [2, 4, 0],
+//       [3, 5, 6]
+//      ]
+// %b = [
+//       [2, 0, 0],
+//       [4, 8, 0],
+//       [6, 10, 12]
+//      ]
```

Comment:
How about saying `[1.0, 0.0, 0.0]` etc, so that the commented values parse as floats by MLIR? Same for the result.